python_modules/libraries/dagster-databricks/dagster_databricks/pipes.py

```diff
@@ -160,6 +183,42 @@ def run(
 
         return PipesClientCompletedInvocation(pipes_session)
 
+    def _enrich_submit_task_dict(
+        self, context: OpExecutionContext, session: PipesSession, submit_task_dict: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        if "existing_cluster_id" in submit_task_dict:
+            # we can't set env vars on an existing cluster
+            # so we must use CLI to pass Pipes params
+            cli_args = session.get_bootstrap_cli_arguments()  # this is a mapping
+
+            for task_type in self.get_task_fields_which_support_cli_parameters():
+                if task_type in submit_task_dict:
+                    existing_params = submit_task_dict[task_type].get("parameters", [])
+
+                    # merge the existing parameters with the CLI arguments
+                    for key, value in cli_args.items():
+                        existing_params.extend([key, value])
```

Comment:
Since most CLI interfaces use the form `--opt1 val --opt2 val posarg1 posarg2...` I wonder if it is better to inject the CLI args at the beginning of the param list rather than the end, so that the pipes CLI args are grouped with other `--` options.

Reply:
good point

Reply:
>Perhaps we could expose a setting controlling the insertion index (1 by default).

Yes good idea

Reply:
On second thought I don't think we need to add this param, just update the pipes databricks to explain what's going on. Users are free to strip out the args if they want in their script.